plugin/pkg/shoot/validator/admission.go

```diff
@@ -885,3 +906,47 @@ func ensureMachineImage(oldWorkers []core.Worker, worker core.Worker, images []c
 
 	return getDefaultMachineImage(images, imageName)
 }
+
+func (v ValidateShoot) validateShootedSeed(a admission.Attributes, shoot, oldShoot *core.Shoot) error {
+	if shoot.Namespace != gardenNamespace {
+		return nil
+	}
+
+	oldVal, oldOk := oldShoot.Annotations[useAsSeedAnnotation]
+	if !oldOk {
+		return nil
+	}
+	if len(oldVal) == 0 {
+		return nil
+	}
+
+	val, ok := shoot.Annotations[useAsSeedAnnotation]
+	if !ok || len(val) == 0 {
```

Comment:
```suggestion
	if ok {
	  return nil
	}
```

Eliminates one level of nesting